drivers/overlay/overlay.go

```diff
@@ -97,18 +100,41 @@ func (d *driver) configure() error {
 			d.store, err = datastore.NewDataStore(datastore.GlobalScope, cfg)
 			if err != nil {
 				err = fmt.Errorf("failed to initialize data store: %v", err)
-				return
 			}
 		}
-
-		d.vxlanIdm, err = idm.New(d.store, "vxlan-id", vxlanIDStart, vxlanIDEnd)
-		if err != nil {
-			err = fmt.Errorf("failed to initialize vxlan id manager: %v", err)
-			return
-		}
 	})
 
-	return err
+	if err != nil {
+		return err
+	}
+
+	if d.store == nil {
+		return types.NoServiceErrorf("datastore is not available")
+	}
+
+	if d.vxlanIdm == nil {
+		return d.initializeVxlanIdm()
+	}
+
```

Comment:
Why is this code change needed ? The existing change seems to do same thing (you could replace the idm initialization with initlaizeVxlanIdm. But beyond that, I dont understand the reason for this code readjustment.